### PR TITLE
Update engines.node to 18; Workflows to 20.

### PR DESCRIPTION
**Pull request by BigBlueHat**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
     timeout-minutes: 10
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x]
+        node-version: [20.x]
```
Comment (JSAssassin) [inline .github/workflows/main.yml:27]: test-node must execute on all versions >= 18.x
```suggestion
        node-version: [18.x, 20.x]
```

**Review by JSAssassin — CHANGES_REQUESTED** [review]
`engines.node` field in `package.json` should be updated to `>=18` and while we're here, can you also please help update `checkout` and `setup-node` actions to the latest versions `@v4`.

**Comment by BigBlueHat** [comment]
Weird...I had the `engines.node` change locally and it somehow got missed when staging the change. I'll make these other changes and submit that also.

**Review by JSAssassin — APPROVED** [review]
Thanks!

**Pull request merged**